index.js

```diff
@@ -75,7 +75,7 @@ function focusTrap(element, userOptions) {
 
     state.active = true;
     state.paused = false;
-    state.nodeFocusedBeforeActivation = doc.activeElement;
+    state.nodeFocusedBeforeActivation = getReturnFocusNode(doc.activeElement);
```

Comment:
Hmm, however shouldn't it be more obvious for `getReturnFocusNode` to be called in the moment when focus trap is being deactivated, instead of when focus trap is being activated? (See line 117)

Reply:
Yeap. I was thinking as the `setReturnFocus` overwrites the returning element, then it does not matter. But, you're right, in case some event (if not now, maybe in future) needs `state.nodeFocusedBeforeActivation` before deactivation, then it makes sense to do it during deactivation.

I will change that.